### PR TITLE
Sebchw upgrading pipeline

### LINKED ISSUE #32: Improve configuration system
Configuration system utilizyng hydra was introduced in https://github.com/SebChw/audio-research-toolkit/pull/24. However, it still lacks many functionalities. I selected just most important ones. In this [tutorial](https://www.kdnuggets.com/2023/03/hydra-configs-deep-learning-experiments.html) and especially [repo](https://github.com/gorodnitskiy/yet-another-lightning-hydra-template). There are many more cool use cases of this system So this should be improved. If you have any idea on improvement create PR and link it here.

**Don't close this issue**

### PR DESCRIPTION
**Pull request by SebChw**
closes #32 

Few improvements and changes introduced:
* I removed BatchSampler from one dataloader as It turned out it is not necessary for torch > 1.12.1. Read about it [here](https://huggingface.co/docs/datasets/use_with_pytorch). I suggest all of you doing these in the rest of datamodules.
* I used some hydra interpolation possibilities
* I added an option to overfit one batch before training. Just for sanity check.
* There was small problem with logging and TorchMetrics which were resetting unexpectedly. It turned out that if you have 3 different splits you should use different metric for each. Check common pitfalls [here](https://torchmetrics.readthedocs.io/en/stable/pages/lightning.html)
* I added custom lightning checkpoint. Now you can save your models wherever you want and also specify name of checkpoint file + define which metric you want to look after.
* Also more things are now uploaded to neptune (config, hydra logs and best model)

### CODE REVIEW

**Review by kordc — APPROVED** [review]
LGTM
